src/Microsoft.AspNetCore.Routing.Abstractions/IEndpointMatchConstraint.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace Microsoft.AspNetCore.Routing
+{
+    public interface IEndpointMatchConstraint
```

Comment:
`IEndpointMatchConstraint` is perilously close to `IEndpointConstraint`. I think one of the them is going to need a different name, otherwise devs will be very confused.